server/use_cases/add_service_commitments.py

```diff
@@ -46,3 +75,99 @@ def check_time_overlap(shifts):
+    def test_valid_shift_commitment():
+        shifts_repo = Mock()
+        commitments_repo = Mock()
+
+        valid_shift = Mock()
+        valid_shift.get_id.return_value = (
+            "mocked-valid-shift-id"
+        )
+        shifts_repo.get_shifts.return_value = [valid_shift]
+
+        valid_commitment = Mock()
+        valid_commitment.service_shift_id = "mocked-valid-shift-id"
+
+        valid_commitment.to_dict.return_value = (
+            {"_id": "commitment-id"}
+        )
+
+        result = add_service_commitments(commitments_repo, shifts_repo, (
+            [valid_commitment])
+        )
+
+        assert result[0]["success"] is True
+        assert result[0]["service_commitment_id"] == "commitment-id"
+        print(f"Valid shift ID -> {result}\n")
+
+    # invalid shift ID (unsuccessful case)
+    def test_invalid_shift_commitment():
+        shifts_repo = Mock()
+        commitments_repo = Mock()
+
+        shifts_repo.get_shifts.return_value = []
+
+        invalid_commitment = Mock()
+        invalid_commitment.service_shift_id = "this-id-does-not-exist"
+
+        invalid_commitment.to_dict.return_value = (
+            {"_id": "commitment-id"}
+        )
+        result = add_service_commitments(commitments_repo, shifts_repo, (
+            [invalid_commitment])
+        )
+
+        assert result[0]["success"] is False
+        assert result[0]["message"] == (
+            "cannot commit to non-existing shift this-id-does-not-exist"
+        )
+        print(f"Invalid shift ID -> {result}\n")
+
+    # mixed valid and invalid shift IDs
+    def test_mixed_shifts_commitment():
+        shifts_repo = Mock()
+        commitments_repo = Mock()
+
+        valid_shift = Mock()
+        valid_shift.get_id.return_value = (
+            "mocked-valid-shift-id"
+        )
+        shifts_repo.get_shifts.return_value = [valid_shift]
+
+        valid_commitment = Mock()
+        valid_commitment.service_shift_id = (
+            "mocked-valid-shift-id"
+        )
+        invalid_commitment = Mock()
+        invalid_commitment.service_shift_id = (
+            "this-id-does-not-exist"
+        )
+        valid_commitment.to_dict.return_value = (
+            {"_id": "valid-commitment-id"}
+        )
+        invalid_commitment.to_dict.return_value = (
+            {"_id": "invalid-commitment-id"}
+        )
+
+        result = add_service_commitments(commitments_repo, shifts_repo, (
+            [valid_commitment, invalid_commitment])
+        )
+
+        assert result[0]["success"] is True
+        assert result[0]["service_commitment_id"] == "valid-commitment-id"
+        assert result[1]["success"] is False
+        assert result[1]["message"] == (
+            "cannot commit to non-existing shift this-id-does-not-exist"
+        )
+        print(f"Mixed shifts -> {result}\n")
+
+    # run all tests
+    test_valid_shift_commitment()
+    test_invalid_shift_commitment()
+    test_mixed_shifts_commitment()
```

Comment:
@emrameduseljac, please move the test code to `server/tests/use_cases/add_service_commitments.py`.